### PR TITLE
Make CreateVirtualParticles use GPU functions for aggregation_type=None

### PR DESCRIPTION
**Pull request by jmsexton03**
## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [x] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

File: Src/Particle/AMReX_ParticleContainerI.H [hunk 1]
```diff
@@ -646,6 +646,36 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>::Re
   }
 }
 
+struct TransformerVirt
+{
+    /**
+       \brief This copies the particle but replaces the id with VirtualParticleID
+       *
+     **/
+    TransformerVirt ()
```
Comment (atmyers) [inline Src/Particle/AMReX_ParticleContainerI.H]: I think you can remove this and let the compiler generate this for you.

File: Src/Particle/AMReX_ParticleContainerI.H [hunk 2]
```diff
@@ -646,6 +646,36 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>::Re
...
+    {}
+
+    template <typename DstData, typename SrcData>
+    AMREX_GPU_HOST_DEVICE
+    void operator() (DstData& dst, const SrcData& src,
+                     int src_i, int dst_i) const noexcept
+    {
+        dst.m_aos[dst_i] = src.m_aos[src_i];
```
Comment (atmyers) [inline Src/Particle/AMReX_ParticleContainerI.H]: I think you can save some code here by using the `copyParticle` function in `AMReX_ParticleTransformation.H`.

File: Src/Particle/AMReX_ParticleContainerI.H [hunk 3]
```diff
@@ -646,6 +646,36 @@ ParticleContainer<NStructReal, NStructInt, NArrayReal, NArrayInt, Allocator>::Re
...
+{
+    /**
+       \brief This copies the particle but replaces the id with VirtualParticleID
+       *
+     **/
+    TransformerVirt ()
+    {}
+
+    template <typename DstData, typename SrcData>
+    AMREX_GPU_HOST_DEVICE
```
Comment (atmyers) [inline Src/Particle/AMReX_ParticleContainerI.H]: `AMREX_FORCE_INLINE`?

File: Src/Particle/AMReX_ParticleContainerI.H (hunk 1 above)
Comment (jmsexton03) [reply Src/Particle/AMReX_ParticleContainerI.H]: Pushed a commit which removes this

File: Src/Particle/AMReX_ParticleContainerI.H (hunk 2 above)
Comment (jmsexton03) [reply Src/Particle/AMReX_ParticleContainerI.H]: Pushed a commit which does this

File: Src/Particle/AMReX_ParticleContainerI.H (hunk 3 above)
Comment (jmsexton03) [reply Src/Particle/AMReX_ParticleContainerI.H]: Pushed a commit which adds the inline, and updates other inlines in Src/Particles for consistency

File: Src/Particle/AMReX_ParticleContainerI.H (hunk 2 above)
Comment (atmyers) [reply Src/Particle/AMReX_ParticleContainerI.H]: Nice - not needed for this PR, but you could also update the ghost one to do the same thing.

Comment (jmsexton03) [reply Src/Particle/AMReX_ParticleContainerI.H]: I will include that update as well (got confused on doing both removing the constructor and updating the copy)

**Review by atmyers — APPROVED** [review]

**Pull request merged**